fff

```diff
@@ -368,6 +369,50 @@ trash() {
     fi
 }
 
+bulk_rename() {
+    # Bulk rename files using '$EDITOR'.
+    rename_file="${XDG_CACHE_HOME:=${HOME}/.cache}/fff/bulk_rename"
+    marked_files=("${@:1:$#-1}")
+
+    # Save marked files to a file and open them for editing.
+    printf '%s\n' "${marked_files[@]}" > "$rename_file"
+    "${VISUAL:-${EDITOR:-vi}}" "$rename_file"
+
+    # Read the renamed files to an array.
+    IFS=$'\n' read -d "" -ra changed_files < "$rename_file"
+
+    # If the user deleted a line, stop here.
+    ((${#marked_files[@]} != ${#changed_files[@]})) && {
+        rm "$rename_file"
+        cmd_line "warn: Line mismatch in rename file. Doing nothing."
```

Comment:
For me, since it causes the function to abort, it's an error type of message, not warning

Reply:
Will do.